xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java

```diff
@@ -0,0 +1,241 @@
+            Map<String, Object> stubMap = FlatMapUtils.flatten(stubBody);
+            Map<String, Object> currentBodyMap = FlatMapUtils.flatten(currentBody);
+            if (!currentBodyMap.keySet().containsAll(stubMap.keySet())) {
+                log.warn("{} stub {} body is different from the aspsp {}", aspsp.getName(), payloadType, payloadType);
+                String changes = aspsp.getName() + ":" + resolver.name() + ":" + payloadType + "-payload";
+                return Optional.of(changes);
+            }
+        } catch (JsonProcessingException e) {
+            log.error("Can't get differences for the {} or wiremock stub body", payloadType, e);
+        }
+        return Optional.empty();
+    }
+
+    private enum PayloadType {
+        REQUEST, RESPONSE;
+    }
+
+    private <T> String getResponseBody(Response<T> response) throws JsonProcessingException {
+        String body = null;
+        T bodyObject = response.getBody();
+        if (bodyObject instanceof String) {
+            body = (String) bodyObject;
+        } else {
+            body = objectMapper.writeValueAsString(bodyObject);
+        }
+        return body;
+    }
+
+    @SuppressWarnings("unchecked")
+    private Optional<String> getResponseBody(Map<String, Object> jsonFile, String adapterName) {
+        Map<String, Object> response = (Map<String, Object>) jsonFile.get(RESPONSE);
+        if (response.containsKey(BODY)) {
+            return Optional.ofNullable((String) response.get(BODY));
+        }
+        if (response.containsKey(BODY_FILE_NAME)) {
+            try {
+                String fileName = (String) response.get(BODY_FILE_NAME);
+                String filePath = adapterName + File.separator + "__files" + File.separator + fileName;
+                InputStream is = getClass().getResourceAsStream(File.separator + filePath);
+                return Optional.ofNullable(IOUtils.toString(is));
+            } catch (IOException e) {
+                log.error("Can't get stub response file", e);
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<String> analyzeRequestBody(WiremockFileResolver resolver, Request.Builder builder, String requestBody) {
+        return analyzePayloadStructure(resolver, requestBody, builder.body(), PayloadType.REQUEST);
+    }
+
+    private Optional<String> analyzeRequestHeaders(WiremockFileResolver resolver, Request.Builder builder, Map<String, Object> requestHeaders) {
+        if (requestHeaders != null && !builder.headers().keySet().containsAll(requestHeaders.keySet())) {
+            log.warn("{} stub headers are different from the request", aspsp.getName());
+            String changes = aspsp.getName() + ":" + resolver.name() + ":request-headers";
+            return Optional.of(changes);
+        }
+        return Optional.empty();
+    }
+
+    private Optional<String> analyzeRequestUrl(WiremockFileResolver resolver, Request.Builder builder, String requestUrl) {
```

Comment:
We removed URL comparison because URL has a lot of unique path parameters but if you think that such comparison will be useful we can create a sub-task for that functionality.

Reply:
I think it is important

Reply:
The sub-task was created https://jira.adorsys.de/browse/XS2AAD-708